edition = "2023";

package graph_mining.in_memory;

option java_multiple_files = true;

message CliqueAggregatorConfig {
  // The minimum density of the clique containers to be returned.
  double min_density = 1;

  // If set, the algorithm may output clusters that are not connected. The
  // clusters still have density >= `min_density`.
  bool allow_disconnected_clusters = 2;

  //////////////////////////////////////////////////////////////////////////////
  // The parameters below are only used for performance testing.
  // We use large tag numbers, as they are likely to be removed in the future.
  //////////////////////////////////////////////////////////////////////////////

  // If false, Bron-Kerbosch pruning is not used and the algorithm may output
  // clusters that are subsets of other clusters.
  bool bron_kerbosch_pruning = 100 [default = true];

  // If set, we disable the use of `BitSetGraph` in the recursive calls to
  // `CliqueAggregator`.
  bool use_gbbs_graph_recursively = 101;
}
